File: vbc/backend/src/main/java/com/vbc/entity/Device.java
package com.vbc.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("vbc_device")
public class Device {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String deviceName;
    private String deviceCode;
    private String ipAddress;
    private String streamUrl;
    private String location;
    private String status;
    private LocalDateTime lastHeartbeat;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: vbc/frontend/src/types/device.ts
export interface Device {
  id: number
  deviceName: string
  deviceCode: string
  ipAddress: string
  streamUrl: string
  location: string
  status: 'ONLINE' | 'OFFLINE' | 'MAINTENANCE'
  lastHeartbeat: string
  createdAt: string
  updatedAt: string
}
